js/libs/builder.js

```diff
@@ -191,21 +196,127 @@ var app = new Vue({
+            runOn: ["insert", "update"],
+            type: "email",
+            payload: $vm.settings.emailTemplate
+          };
+
+          Fliplet.DataSources.getById($vm.settings.dataSourceId).then(function(dataSource) {
+            if (dataSource.hooks.length) {
+              // Update existing hook
+              var currentHook = _.find(dataSource.hooks, function(o) {
+                return o.widgetInstanceId == $vm.settings.id;
+              });
+
+              currentHook.payload = $vm.settings.emailTemplate;
+
+              var index = _.findIndex(dataSource.hooks, function(o) {
+                return o.widgetInstanceId == $vm.settings.id
+              });
+              dataSource.hooks.splice(index, 1, currentHook);
+
+              Fliplet.DataSources.update($vm.settings.dataSourceId, {
+                hooks: dataSource.hooks
+              });
+            } else {
+              // Add new hook
+              dataSource.hooks.push(newHook);
+              Fliplet.DataSources.update($vm.settings.dataSourceId, {
+                hooks: dataSource.hooks
+              });
+            }
+          });
+        }
+
+        Fliplet.Widget.autosize();
+      });
+    },
+    configureGenerateEmailDevice: function() {
+      var $vm = this;
+      var allFields = $vm.settings.fields;
+      // Creates default email template
+      var defaultEmailTemplate = '<h1>' + $vm.settings.displayName + '</h1><p>A form submission has been received.</p>';
+      defaultEmailTemplate += '<ul>'
+      for (var i = 0; i < allFields.length; i++) {
+        if (typeof allFields[i]._submit === "undefined" || allFields[i]._submit) {
+          defaultEmailTemplate += '<li style="line-height: 24px;">' + allFields[i].label + ': {{' + allFields[i].name + '}}</li>';
+        }
+      }
+      defaultEmailTemplate += '</ul>'
+
+      var defaultEmailSettings = {
+        subject: 'Form entries from "' + $vm.settings.displayName + '" form',
+        html: defaultEmailTemplate,
+        to: []
+      };
+      var emailProviderData = ($vm.settings && $vm.settings.generateEmailTemplate) || defaultEmailSettings;
+
+      window.generateEmailProvider = Fliplet.Widget.open('com.fliplet.email-provider', {
+        data: emailProviderData
+      });
+
+      window.generateEmailProvider.then(function onForwardEmailProvider(result) {
+        window.generateEmailProvider = null;
+        $vm.settings.generateEmailTemplate = result.data;
+        Fliplet.Widget.autosize();
+      });
     }
   },
   watch: {
     'dataSources': function(newVal) {
-      changeSelectText()
+      changeSelectText();
     },
     'permissionToChange': function(newVal) {
       Fliplet.Widget.toggleSaveButton(newVal);
     },
     'isAddingFields': function(newVal) {
       if (newVal) {
         Fliplet.Studio.emit('widget-mode', 'wide');
-        setTimeout(() => {
-          attatchObservers()
-        }, 1)
+        setTimeout(function() {
```

Comment:
@tonytlwu because we don't compile these files with babel, hence using ES6 might fail where it's not supported to I asked to revert to "ol' plain ES5".